src/transport.rs

```diff
@@ -139,20 +140,24 @@ macro_rules! implement_http_transport {
 
         impl $typename {
             /// Creates a new transport.
-            pub fn new(options: &ClientOptions) -> $typename {
+            pub fn new(options: &ClientOptions, $hc_client: Option<$hc_client_ty>) -> $typename {
```

Comment:
I would prefer if we add a `with_client` constructor that takes a client and leave `new` as it. That way upgrading is easier and also makes for a slightly nicer api. They can both call into an internal helper method.

Reply:
Ok, I will update the code.